shared_python/Args.py

```diff
@@ -11,15 +12,11 @@ def _load_args_from_file(filepath):
   props = {}
   sep = ':'
   with open(filepath, "rt") as f:
-    for line in f:
-      l = line.strip()
-      if l:
-        key_value = l.split(sep)
-        key = key_value[0].strip()
-        value = sep.join(key_value[1:]).strip().strip('"')
-        props[key] = value
-  return props
-
+    try:
+      x = yaml.safe_load(f)
+      return x
+    except yaml.YAMLError as exc:
+      log.error("YAML parsing failure: {}".format(exc))
```

Comment:
Is this just swallowing the error and returning a `None` that will leap out and surprise us later? Or does it just bomb out at this point (which is probably preferable)?

Reply:
Yes, that was the idea, as that was what we were doing before.  (Swallowing the error).  I agree that throwing an exception is better practice, though, and if you're willing to let us go that way, we should!